tests/unit_tests/simpll/DifferentialFunctionComparatorTest.cpp

```diff
@@ -0,0 +1,1159 @@
+                                     Type::getInt8Ty(CtxR)),
+              0);
+    ASSERT_EQ(DiffComp->testCmpTypes(ArrayType::get(Type::getInt8Ty(CtxL), 10),
+                                     ArrayType::get(Type::getInt8Ty(CtxR), 11)),
+              0);
+    // Boolean type should stay unequal.
+    ASSERT_EQ(DiffComp->testCmpTypes(ArrayType::get(Type::getInt1Ty(CtxL), 10),
+                                     ArrayType::get(Type::getInt8Ty(CtxR), 11)),
+              1);
+}
+
+/// Tests whether calls are properly marked for inlining while comparing
+/// basic blocks.
+TEST_F(DifferentialFunctionComparatorTest, CmpBasicBlocksInlining) {
+    // Create the basic blocks.
+    BasicBlock *BBL = BasicBlock::Create(CtxL, "", FL);
+    BasicBlock *BBR = BasicBlock::Create(CtxR, "", FR);
+
+    // Create auxilliary functions to inline.
+    Function *AuxFL = Function::Create(
+            FunctionType::get(
+                    Type::getVoidTy(CtxL), {Type::getInt32Ty(CtxR)}, false),
+            GlobalValue::ExternalLinkage,
+            "AuxFL",
+            &ModL);
+    Function *AuxFR = Function::Create(
+            FunctionType::get(
+                    Type::getVoidTy(CtxR), {Type::getInt32Ty(CtxR)}, false),
+            GlobalValue::ExternalLinkage,
+            "AuxFR",
+            &ModR);
+
+    // Test inlining.
+    CallInst *CL =
+            CallInst::Create(AuxFL->getFunctionType(),
+                             AuxFL,
+                             {ConstantInt::get(Type::getInt32Ty(CtxL), 5)},
+                             "",
+                             BBL);
+    CallInst *CR =
+            CallInst::Create(AuxFR->getFunctionType(),
+                             AuxFR,
+                             {ConstantInt::get(Type::getInt32Ty(CtxR), 6)},
+                             "",
+                             BBR);
+    ReturnInst::Create(CtxL, BBL);
+    ReturnInst::Create(CtxR, BBR);
+
+    ASSERT_EQ(DiffComp->testCmpBasicBlocks(BBL, BBR), 1);
+    std::pair<const CallInst *, const CallInst *> expectedPair{CL, CR};
+    ASSERT_EQ(ModComp->tryInline, expectedPair);
+}
+
+/// Tests ignoring of instructions that don't cause a semantic difference in
+/// cmpBasicBlocks.
+/// Note: the functioning of mayIgnore is tested in the test for cmpValues.
+TEST_F(DifferentialFunctionComparatorTest, CmpBasicBlocksIgnore) {
+    BasicBlock *BBL = BasicBlock::Create(CtxL, "", FL);
+    BasicBlock *BBR = BasicBlock::Create(CtxR, "", FR);
+
+    new AllocaInst(Type::getInt8Ty(CtxL), 0, "var", BBL);
+    new AllocaInst(Type::getInt8Ty(CtxR), 0, "var1", BBR);
+    new AllocaInst(Type::getInt8Ty(CtxR), 0, "var2", BBR);
+    ReturnInst::Create(CtxL, BBL);
+    ReturnInst::Create(CtxR, BBR);
+
+    ASSERT_EQ(DiffComp->testCmpBasicBlocks(BBL, BBR), 0);
+    ASSERT_EQ(DiffComp->testCmpBasicBlocks(BBR, BBL), 0);
+}
+
+/// Tests the comparison of global values.
+TEST_F(DifferentialFunctionComparatorTest, CmpGlobalValues) {
```

Comment:
This test could be split into several smaller tests. I suggest doing so, since it will improve the developer experience - if a test fails, the developer needs to read and understand a short test only.